cmake/mkldnn_omp.diff

```diff
@@ -1,12 +1,28 @@
-diff --git a/mkldnn/src/cmake/OpenMP.cmake b/../../cmake/OpenMP.cmake
-index f9c3620e..4e873dab 100644
---- a/mkldnn/src/cmake/OpenMP.cmake
-+++ b/../../cmake/OpenMP.cmake
-@@ -30,7 +30,7 @@ if (APPLE AND CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+diff --git a/cmake/OpenMP.cmake b/cmake/OpenMP.cmake
+index f9c3620e..ed1ad6c2 100644
+--- a/cmake/OpenMP.cmake
++++ b/cmake/OpenMP.cmake
+@@ -30,14 +30,19 @@ if (APPLE AND CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
      # But we still want to build the library.
      set(_omp_severity "WARNING")
  else()
 -    set(_omp_severity "FATAL_ERROR")
 +    set(_omp_severity "WARNING")
  endif()
 
+
+ macro(forbid_link_compiler_omp_rt)
+     if (NOT WIN32)
+-        set_if(OpenMP_C_FOUND CMAKE_C_CREATE_SHARED_LIBRARY_FORBIDDEN_FLAGS ${OpenMP_C_FLAGS})
+-        set_if(OpenMP_CXX_FOUND CMAKE_CXX_CREATE_SHARED_LIBRARY_FORBIDDEN_FLAGS ${OpenMP_CXX_FLAGS})
++        if (OpenMP_C_FOUND)
```

Comment:
Why is this change needed? Macro set_if expands to the same thing.

Reply:
`set_if` doesn't support empty variables. i.e if OpenMP_CXX_FOUND was never defined, it would not return false. And I was getting following error. 
```Could NOT find OpenMP (missing:  OpenMP_CXX_FLAGS)
....
CMake Error at cmake/OpenMP.cmake:40 (set_if):
  set_if Macro invoked with incorrect arguments for macro named: set_if
Call Stack (most recent call first):
  cmake/OpenMP.cmake:57 (forbid_link_compiler_omp_rt)```